src/charm.py

```diff
@@ -53,6 +54,10 @@ def __init__(self, *args):
             depends_on=[],
         )
 
+        self.owasp_logger = self.charm_reconciler.add(
+            component=OWASPLoggerComponent(charm=self, stored=self._stored)
+        )
```

Comment:
suggestion: OWASP component should depend on leadership gate component, like all the components do